java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -117,17 +117,17 @@ private void run(@Nullable final ExplodedGraph.Node node, @Nullable final Symbol
       }
       visited.add(currentNode);
 
-      Symbol newTrackSymbol = nodeSymbol.trackSymbol;
+      Symbol newTrackSymbol;
       if (currentNode.programPoint.syntaxTree() != null) {
-        newTrackSymbol = addFlowFromLearnedSymbols(currentNode, newTrackSymbol);
+        newTrackSymbol = addFlowFromLearnedSymbols(currentNode, nodeSymbol.trackSymbol);
         Stream<Constraint> learnedConstraints = addFlowFromLearnedConstraints(currentNode);
         if (learnedConstraints.anyMatch(terminateTraversal)) {
           continue;
         }
+      } else {
+        newTrackSymbol = nodeSymbol.trackSymbol;
       }
-      for (ExplodedGraph.Node parent : currentNode.getParents()) {
-        workList.push(new NodeSymbol(parent, newTrackSymbol));
-      }
+      currentNode.parents().forEach(parent -> workList.push(new NodeSymbol(parent, newTrackSymbol)));
```

Comment:
As far as I can see this is a cosmetic refactoring and should not be  should not be part of this commit.